PKHeX/Legality/Core.cs

```diff
@@ -942,48 +937,54 @@ internal static EncounterTrade getValidIngameTrade(PKM pkm, GameVersion gameSour
 
             EncounterTrade[] table = getEncounterTradeTable(pkm);
 
-            EncounterTrade z = table?.FirstOrDefault(f => p.Any(r => r.Species == f.Species));
+            IEnumerable<EncounterTrade> poss = table?.Where(f => p.Any(r => r.Species == f.Species));
```

Comment:
I recommend making this a list, since you're calling Count() then enumerating through it.  This could have a minor performance benefit depending on whether or not Linq stores the results of the first enumeration.
```
var poss = table?.Where(f => p.Any(r => r.Species == f.Species)).ToList();
```